src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1549,7 +1549,7 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
                     binder = new ExecutableCodeBinder(arrowExpression, sourceMethod, binder);
                     importChain = binder.ImportChain;
                     // Add locals
-                    return binder.BindExpressionBodyAsBlock(arrowExpression, diagnostics);
+                    return binder.WithPatternVariablesIfAny(arrowExpression.Expression).BindExpressionBodyAsBlock(arrowExpression, diagnostics);
```

Comment:
Do we ever bind an expression body without the pattern variables? I'm wondering if we could fold this logic directly into the `BindExpressionBodyAsBlock` helper.